cpp/src/tests/db/find_matches_test.cu

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gtest/gtest.h"
+#include "high_res_clock.h"
+#include <cugraph.h>
+#include "test_utils.h"
+#include "db/db_operators.cuh"
+#include "utilities/graph_utils.cuh"
+
+class Test_FindMatches: public ::testing::Test {
+public:
+  Test_FindMatches() {}
+  virtual void SetUp() {
+    cugraph::db_pattern<int32_t> p;
+    cugraph::db_pattern_entry<int32_t> p1(0);
+    cugraph::db_pattern_entry<int32_t> p2(1);
+    cugraph::db_pattern_entry<int32_t> p3(2);
+    p.addEntry(p1);
+    p.addEntry(p2);
+    p.addEntry(p3);
+    table.addColumn("Start");
+    table.addColumn("EdgeType");
+    table.addColumn("End");
+    table.addEntry(p);
+    table.flush_input();
+  }
+  virtual void TearDown() {}
+  void insertConstantEntry(int32_t a, int32_t b, int32_t c) {
+    cugraph::db_pattern<int32_t> p;
+    cugraph::db_pattern_entry<int32_t> p1(a);
+    cugraph::db_pattern_entry<int32_t> p2(b);
+    cugraph::db_pattern_entry<int32_t> p3(c);
+    p.addEntry(p1);
+    p.addEntry(p2);
+    p.addEntry(p3);
+    table.addEntry(p);
+  }
+  cugraph::db_table<int32_t> table;
+};
+
+TEST_F(Test_FindMatches, verifyIndices) {
+  int32_t* offsets_d = (int32_t*)table.getIndex(0).getOffsets()->data;
+  int32_t offsetsSize = table.getIndex(0).getOffsets()->size;
+  int32_t* indirection_d = (int32_t*)table.getIndex(0).getIndirection()->data;
+  int32_t indirectionSize = table.getIndex(0).getIndirection()->size;
+  int32_t* offsets_h = (int32_t*)malloc(sizeof(int32_t) * offsetsSize);
```

Comment:
Shouldn't we use new/delete rather than malloc/free?  It's a bit more type safe and a little cleaner in the coding also.  Might even consider using a std::vector and letting the STL take care of the memory management for you.

Reply:
Changed to use new/delete rather than malloc/free.